openmdao/core/tests/test_distrib_adder.py

```diff
@@ -89,7 +89,8 @@ def test_distributed_adder(self):
 
         prob.model.add_subsystem('des_vars', IndepVarComp('x', np.ones(size)), promotes=['x'])
         prob.model.add_subsystem('plus', DistributedAdder(size=size), promotes=['x', 'y'])
-        summer = prob.model.add_subsystem('summer', Summer(size=size), promotes=['y', 'sum'])
+        summer = prob.model.add_subsystem('summer', Summer(size=size), promotes_outputs=['sum'])
+        prob.model.promotes('summer', inputs=['y'], src_indices=slicer[:])
```

Comment:
Is this test modification just so that you can also test 'y' with the promotes method, or is there some reason it didn't work anymore when done in add_subsystem?

Reply:
Since the default behavior has changed when we connect a distrib output to a serial input, to get the same behavior as before, i.e. the serial variable is sized according to the full distrib size of the output, you have to set src_indices for that input to slicer[:] (or some equivalent slice or index array).